Add HTML text formatting option

eseiver opened the pull request:
Decided to address #191 myself! This changes `_TranscriptParser()` to have a `preserve_formatting` option that defaults to `False`. I then added the ability to set this parameter at the user level in `YouTubeTranscriptApi.get_transcript()`, making changes to the intermediary classes and methods where appropriate. Finally, I updated the testing XML asset in [transcript.xml.static](https://github.com/eseiver/youtube-transcript-api/blob/html_formatting/youtube_transcript_api/test/assets/transcript.xml.static) to include escaped italics, as well as adding a new test `test_get_transcript_formatted()` that keeps the italics. None of the other similar tests were changed because it still defaults to the old version.

jdepoix commented on youtube_transcript_api/_api.py:
```diff
@@ -109,7 +110,7 @@ def get_transcripts(cls, video_ids, languages=('en',), continue_after_error=Fals
 
     @classmethod
-    def get_transcript(cls, video_id, languages=('en',), proxies=None, cookies=None):
+    def get_transcript(cls, video_id, languages=('en',), proxies=None, cookies=None, preserve_formatting=False):
```
Please add `preserve_formatting` to the docstrings

jdepoix commented on youtube_transcript_api/_api.py:
```diff
@@ -68,7 +68,8 @@ def list_transcripts(cls, video_id, proxies=None, cookies=None):
+                                                            preserve_formatting=preserve_formatting)
 
     @classmethod
     def get_transcripts(cls, video_ids, languages=('en',), continue_after_error=False, proxies=None, cookies=None):
```
`preserve_formatting` should also be added to `get_transcripts`

jdepoix reviewed (changes requested):
Hi @eseiver,
thank you very much for your contribution!!! 🥳 👍 

Sorry that it took so long, but I just didn't get around to having a look at it.

There's only a few minor things and then it should be good to be merged.

eseiver commented:
Thanks so much for this helpful review! I have addressed all the points in some additional commits and updated a few more tests. Let me know if there's anything else needed.

jdepoix commented on youtube_transcript_api/_transcripts.py:
```diff
@@ -314,16 +334,28 @@ def translate(self, language_code):
 class _TranscriptParser(object):
-    HTML_TAG_REGEX = re.compile(r'<[^>]*>', re.IGNORECASE)
+    def __init__(self, preserve_formatting=False):
+        self.preserve_formatting = preserve_formatting
```
We do not need to keep this state around, since it is not needed after the object has been initialized

jdepoix reviewed (changes requested):
Hi @eseiver,
thank you very much for taking the time to implement the requested changes. Unfortunately, I just now realized that we don't need to hold the state for `preserve_formatting` in that many places, as it does not become relevant until the actual transcript is fetched. However, implementing that change should mainly require deleting code and should actually make the code simpler 👍

jdepoix commented:
Somehow the builds haven't run yet, I will have a look into that!

eseiver commented:
Alright that makes sense that `preserve_formatting` can be mostly in class methods not in `__init__`. Updated and also moved `_FORMATTING_TAGS` into `_TranscriptParser`.

jdepoix commented:
Hi @eseiver, thanks for the additional fixes! I just added a few of the missing things myself. I will merge this now and release it with v0.6.0 🥳

jdepoix commented:
I just release this in [v0.6.0](https://github.com/jdepoix/youtube-transcript-api/releases/tag/v0.6.0). Thank you for your contribution! 😊 🙏
